### PR TITLE
Add application version filter

### PR DESCRIPTION
**Pull request by islean**
## Description

In /options the applications returned all have versions. This PR adds such a filter to the new /applications/order_type. Closes https://github.com/Clinical-Genomics/add-new-tech/issues/123

### Added

-

### Changed

- /applications/order_type only returns applications which have versions

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/store/crud/read.py [hunk 1]
```diff
@@ -822,15 +823,19 @@ def get_applications(self) -> list[Application]:
             .all()
         )
 
-    def get_active_applications_by_order_type(self, order_type: str) -> list[Application]:
+    def get_active_applications_by_order_type(self, order_type: OrderType) -> list[Application]:
         """
         Return all possible non-archived applications for an order type.
```
Comment (diitaz93) [inline cg/store/crud/read.py:828]: update docstring

File: cg/store/filters/status_application_filters.py [hunk 2]
```diff
@@ -27,6 +27,10 @@ def filter_applications_is_not_external(applications: Query, **kwargs) -> Query:
     return applications.filter(Application.is_external == False)
 
 
+def filter_applications_has_versions(applications: Query, **kwargs) -> Query:
```
Comment (diitaz93) [inline cg/store/filters/status_application_filters.py:30]: Would be nice a unit test for the filter function

**Review by diitaz93 — APPROVED** [review]
Great 💯

File: cg/store/filters/status_application_filters.py (hunk 2 above)
Comment (islean) [reply cg/store/filters/status_application_filters.py:30]: Added!

**Comment by islean** [comment]
Deployed to the VMs

**Pull request merged**